src/pngwriter.cc

```diff
@@ -915,9 +915,84 @@ void pngwriter::close()
 		bit_depth_, PNG_COLOR_TYPE_RGB, PNG_INTERLACE_NONE,
 		PNG_COMPRESSION_TYPE_DEFAULT, PNG_FILTER_TYPE_DEFAULT);
 
-   if(filegamma_ < 1.0e-1)
+   png_set_gAMA(png_ptr, info_ptr, filegamma_);
```

Comment:
if you call a non-const member function here, this function can not be `const`, right?

Reply:
Yes, png_set_gAMA() is not a member function